src/Misc/FiniteField.jl

```diff
@@ -12,11 +12,6 @@ function FlintFiniteField(p::fmpz; cached::Bool = true)
   return k, k(1)
 end
 
-function FlintFiniteField(p::Int, k::Int; cached::Bool = true)
-  @assert isprime(p)
-  return FlintFiniteField(p, k, "o", cached = cached)
-end
-
 GF(p::Integer, k::Int, s::AbstractString="o"; cached::Bool = true) = FlintFiniteField(p, k, s, cached = cached)[1]
 GF(p::fmpz, k::Int, s::AbstractString="o"; cached::Bool = true) = FlintFiniteField(p, k, s, cached = cached)[1]
```

Comment:
I was wondering if those can't also go, but they still diverge from their Nemo counterparts:
- Here `s` can be specified, but not in Nemo
- Nemo version call `isprime` resp. `isprobable_prime`
- Nemo only supports `Int` and `UInt` but not arbitrary `Integer`... But actually it is not clear to me how that would work here, given that there is no `FlintFiniteField` taking any `Integer`, such as a `BigInt`?! So perhaps  `GF(p::Integer, ...)` and also the `FlintFiniteField(p::Integer ...)` method at the top of the file should go?

If nothing else, it would be good to do this:
```suggestion
GF(p::Integer, k::Int, s::Union{AbstractString,Symbol}="o"; cached::Bool = true) = FlintFiniteField(p, k, s, cached = cached)[1]
GF(p::fmpz, k::Int, s::Union{AbstractString,Symbol}="o"; cached::Bool = true) = FlintFiniteField(p, k, s, cached = cached)[1]
```